host/switchless_manager.c

```diff
@@ -0,0 +1,172 @@
+/* Copyright (c) Microsoft Corporation.  All rights reserved.
+ * Licensed under the MIT License. */
+
+#include "switchless_manager.h"
+#include <openenclave/internal/calls.h>
+#include "sgx/enclave.h"
+
+/* thread helper functions */
+#if _MSC_VER
+
+#define _thread_return_t DWORD WINAPI
+typedef LPVOID _thread_arg_type;
+#define _THREAD_RETURN_VAL 0
+
+typedef DWORD (*_thread_op_t)(LPVOID);
+
+static int _thread_create(
+    oe_thread_t* thread,
+    _thread_op_t op,
+    _thread_arg_t arg)
+{
+    *thread = CreateThread(NULL, 0, op, arg, 0, NULL);
+    return NULL == *thread;
+} /* _thread_create */
+
+static void _thread_join(_thread_arg_t thread)
```

Comment:
These function would better fit in hostthread.h, windows/hostthread.c, linux/hostthread.c

Reply:
changed.

Reply:
moved.